packages/cli/src/commands/generate/scaffold/__tests__/scaffold.test.js

```diff
@@ -20,8 +20,8 @@ describe('in javascript (default) mode', () => {
     })
   })
 
-  test('returns exactly 17 files', async () => {
-    expect(Object.keys(files).length).toEqual(17)
+  test('returns exactly 19 files', async () => {
```

Comment:
Hey Tobbe, do you think we could add a separate test for:
a) Making sure the correct path to lib/formatters is generated
b) We take a snapshot of the lib/formatters

Reply:
Instead of a snapshot I decided to write actual tests for all the functions in the file https://github.com/redwoodjs/redwood/pull/6221/files#diff-b7cebc9931b5f4cc3f41475724bdcb0ec14d14c3d6c9e4c113ef22b75db96a31

I thought that was enough, or even better. But if you don't agree, let me know why :) 

I'll see what I can do about testing the path

Reply:
Yeah I saw that's great - but the snapshots and specific path tests help us prevent regressions. Its happened a few times with generators.

Reply:
Also relevant: https://github.com/redwoodjs/redwood/pull/6425

Reply:
If you don't generate the tests, because someone changed something (you can't personally review every PR!) - then the tests won't fail. For a while, in generate cell or something, we only had tests that looked at the paths. The paths were there (so the count was still x), but the content of the file was null.

Simple explicit tests is all I'm suggesting - the more complicated and layers we have to dig to notice a regression, the less likely it will be found.

Reply:
Ohh, but they will fail. We run all those tests as part of e2e CI on every PR

Reply:
The test is for CLI - not about the tests you're generating. 

Isn't it odd that I have to wait for an e2e CI suite to run to see if I broke something in the generators if I change something?

Unit tests serve the following purpose imo:
- prevent regressions of your code (in this case, the generator cli)
- documentation to anyone looking at the code
- quick feedback for anyone contributing to this part of the code
- easy to review what's changed when someone changes it

By the logic you're using, technically we don't need the majority of unit tests because the e2e tests would capture all the failures right? 

Again I think we've wasted more time talking about this than it would take to add the two lines ;) needed for the tests - so up to you if you want to add them or not. I'm not going to fight over this

Reply:
Thanks for staying with me here. And your request for testing the path did actually catch a bug, so I'm very grateful for that one :)  

I agree with all your points. I don't think snapshot tests are the best solution/choice for solving those points. But clearly relying on e2e CI tests also aren't the best choice. And at least the snapshots will give faster feedback, so let's go with that!

Reply:
Yeah - snapshots are at best "sort of helpful" - but there've been numerous regressions in generators when we didn't have them, and the path tests (Cell, Sdl, Service, Functions off the top of my head).